### ISSUE TITLE
Not Working in iOS 11

### ISSUE THREAD

**Issue by sudeep23** [corin8823/Popover#88]
Popover works till iOS 10 but it doesn't pop up at all in iOS 11.

**Comment by kramsretep** [comment]
I'm seeing the same thing. One additional detail: it does popup if an input field on the same view is the first responder (and hence the keyboard is showing).

**Comment by chonex143** [comment]
Its now working in iOS 11, please update.

**Comment by corin8823** [comment]
Sorry for replying late.
Example works, but how does it work?

**Comment by chonex143** [comment]
@corin8823 I just try this in your [PR](https://github.com/corin8823/Popover/pull/85) and it works awesome!

i Just change the value from `default` to `ON`

**Comment by chonex143** [comment]
@corin8823 I do some changes on your code and now it works perfectly.

In your Popover.swift `fileprivate func show() {` add a `DispatchQueue.main`

```
fileprivate func show() {
    self.setNeedsDisplay()
    DispatchQueue.main.async {
        switch self.popoverType {
        case .up:
            self.contentView.frame.origin.y = 0.0
        case .down:
            self.contentView.frame.origin.y = self.arrowSize.height
        }
        self.addSubview(self.contentView)
        self.containerView.addSubview(self)
        self.create()
        self.transform = CGAffineTransform(scaleX: 0.0, y: 0.0)
        self.willShowHandler?()
        UIView.animate(withDuration: self.animationIn, delay: 0,
                       usingSpringWithDamping: 0.7,
                       initialSpringVelocity: 3,
                       options: UIViewAnimationOptions(),
                       animations: {
                        self.transform = CGAffineTransform.identity
        }){ _ in
            self.didShowHandler?()
        }
        UIView.animate(withDuration: self.animationIn / 3,
                       delay: 0,
                       options: .curveLinear,
                       animations: { _ in
                        self.blackOverlay.alpha = 1
        }, completion: { _ in
        })
    }
  }
```

**Comment by Vibrosonic** [comment]
I have this issue as well. I'm starting to investigate, but wonder if anyone's got any insight to share?

**Comment by Vibrosonic** [comment]
Okay, for what it's worth, I've discovered that if I switch to use the show() variant that includes "inView", then it works for me consistently:

  show(_ contentView: UIView, fromView: UIView, inView: UIView)

I hope this is helpful.

**Comment by Vibrosonic** [comment]
Please check  https://github.com/corin8823/Popover/issues/82 , where borkuslav reported this much earlier than I did.

**Comment by khantthulinn** [comment]
yes. same. will you guys fix soon? if not, I gona need to find another library since our app launch is very near to support ios 11.  Or is there any temp workaround?

**Comment by khantthulinn** [comment]
I found workaround. I just show like this. I need to specify inView. 

```
        let window = UIApplication.shared.keyWindow!
        self.popover.show(tableView, fromView: fromView, inView: window)

```

**Comment by chonex143** [comment]
Any update?

**Comment by VatsalDev** [comment]
guard let rootView = UIApplication.shared.keyWindow ?? UIApplication.shared.windows.last else {
        return
    }
    self.show(contentView, fromView: fromView, inView: rootView)

Update above code in place of 
    guard let rootView = UIApplication.shared.windows.last ?? UIApplication.shared.keyWindow else {
        return
    }
    self.show(contentView, fromView: fromView, inView: rootView)
at method 
// open func show(_ contentView: UIView, fromView: UIView)

**Comment by CristianCardosoA** [comment]
That method works fine.
open func show(_ contentView: UIView, fromView: UIView) {
guard let rootView = UIApplication.shared.keyWindow ?? UIApplication.shared.windows.last else {
return
}
self.show(contentView, fromView: fromView, inView: rootView)
}

**Comment by paulshapiro** [comment]
imho there should probably be an error log, fatalError, exception raised, or assert(false) on the guarded `return` for visibility

**Comment by CristianCardosoA** [comment]
Update your cocoa pod file. @paulshapiro

**Comment by vrwim-bazookas** [comment]
Fixed in 231633731d691e0ac2e0045f2de055f5aa48c70d & tested on iOS 11.1 beta 5; this issue can be closed @corin8823

**Comment by corin8823** [comment]
thanks 🤓  @vrwim-bazookas
